### ISSUE TITLE
Find/Implement workaround for 'static lifetime of Snapshot instances

### ISSUE THREAD

**Issue by zbuc** [penumbra-zone/penumbra#1512]
Currently we've temporarily implemented our `Snapshot` struct to wrap a `rocksdb::Snapshot` with a `'static` lifetime:

```rust
pub(crate) struct Snapshot {
    // TODO: the `'static` lifetime is a temporary hack and we'll need to find a workaround separately
    rocksdb_snapshot: rocksdb::Snapshot<'static>,
    jmt_version: jmt::Version,
}
```

This was the quickest way to get an implementation underway, since `rocksdb::Snapshot` holds a `&db: DBAccess` which requires a lifetime constraint. However, in the real world, it's unlikely that a `'static` lifetime is actually feasible for us to use here.

We'll need to find some kind of workaround.

Ideas:

1. Fork the Rust rocksdb library and provide a better-shaped API
2. Perform some unsafe pointer wrangling to reborrow a pointer dereference and avoid lifetime checking
3. Use [std::mem::transmute](https://doc.rust-lang.org/std/index.html)::[mem](https://doc.rust-lang.org/std/mem/index.html)::[transmute](https://doc.rust-lang.org/std/mem/fn.transmute.html#) to get a different lifetime
4. At `pd` start, `Box` the db instance (rather than `Arc<...>`) and leak the box so we have a `'static` reference valid for the lifetime of the program

**Comment by redshiftzero** [comment]
This will be closed by #1523 ?

**Comment by zbuc** [comment]
Yes
